pkg/controller/masteruserrecord/masteruserrecord_controller_test.go

```diff
@@ -546,6 +550,55 @@ func TestDeleteMultipleUserAccountsViaMasterUserRecordBeingDeleted(t *testing.T)
 		DoesNotHaveFinalizer()
 }
 
+func TestDisablingMasterUserRecord(t *testing.T) {
+	// given
+	logf.SetLogger(logf.ZapLogger(true))
+	s := apiScheme(t)
+	mur := murtest.NewMasterUserRecord("john", murtest.DisabledMur(true))
+	memberClient := test.NewFakeClient(t)
+	hostClient := test.NewFakeClient(t, mur)
+	cntrl := newController(hostClient, s, newGetMemberCluster(true, v1.ConditionTrue),
+		clusterClient(test.MemberClusterName, memberClient))
+
+	// when
+	_, err := cntrl.Reconcile(newMurRequest(mur))
+	require.NoError(t, err)
+
+	memberClient.MockGet = func(ctx context.Context, key client.ObjectKey, obj runtime.Object) error {
+		console := types.NamespacedName{Namespace: consoleNamespace, Name: "console"}
+		che := types.NamespacedName{Namespace: cheNamespace, Name: "che"}
+		nsdName := namespacedName("toolchain-member-operator", mur.Name)
+		if _, ok := obj.(*routev1.Route); ok && (key == console || key == che) {
+			return nil
+		}
+
+		if key == nsdName {
+			userAccount := obj.(*toolchainv1alpha1.UserAccount)
+			userAccount.Status.Conditions, _ = condition.AddOrUpdateStatusConditions(userAccount.Status.Conditions, toBeDisabled())
+			return nil
+		}
+		return memberClient.Client.Get(ctx, key, obj)
+	}
+
+	memberClient.MockUpdate = func(ctx context.Context, obj runtime.Object, opts ...client.UpdateOption) error {
+		return nil
+	}
```

Comment:
@alexeykazakov maybe it's not needed? when trying to run the test i ran into a problem where this stack trace of funcs is called: `ensureUserAccount()` -> `synchronizeStatus()`-> `withClusterDetails()`. Inside `withClusterDetails()` we attempt to call both `withConsoleURL()` and `withCheDashboardURL()`. So in order to get the test to complete, i mocked these calls. Is there another way of doing this?

Reply:
Just add a console route when you creating a fake client. Thine it will be available in the client which will imitate the actual cluster client: https://github.com/codeready-toolchain/host-operator/blob/d7d62146578a94d447c20228eb702471865c79d3/pkg/controller/masteruserrecord/sync_test.go#L40

Reply:
ohhh! facepalm* i did use this in other tests in member operator, i forgot. thanks :-)

Reply:
OK. But why do you also mock update and update the status in the get?